internal/services/payment_from_submitter_service.go

```diff
@@ -75,7 +75,9 @@ func (s PaymentFromSubmitterService) syncTransaction(ctx context.Context, dbTx d
 	}
 
 	// 4. Set synced_at for the synced transaction
-	err = s.tssModel.UpdateSyncedTransactions(ctx, s.tssModel.DBConnectionPool, []string{transaction.ID})
+	err = db.RunInTransaction(ctx, s.tssModel.DBConnectionPool, nil, func(tssDBTx db.DBTransaction) error {
+		return s.tssModel.UpdateSyncedTransactions(ctx, tssDBTx, []string{transaction.ID})
+	})
```

Comment:
Good question! The point here is that they're **not the same** connection pool! The entire function runs through the SDP Models connection pool. So we **can't** use the `dbTx` from the `sdpModels` to run the `tssModel` methods since they are different schemas. Then we need to run a database transaction specific for the `TSS Connection Pool`. So if any error occurs, we roll back the TSS database transaction. Does that make sense?

Reply:
It makes sense, thanks for explaining 👍